TelemetryDesignDoc.md

```diff
@@ -0,0 +1,327 @@
+# 2022-2023 Telemetry Rewrite Proposal
+
+This is a design document for a proposed refactoring of WPILib telemetry features for the 2022-2023 season.
+
+## Motivations
+
+As WPILib has grown, its telemetry architecture has grown with it.  Heading into the 2022-2023 season, WPILib currently plans to support:
+
+* A communications protocol (NetworkTables)
+* Four dashboards (LabView dashboard, SmartDashboard, Shuffleboard, Glass)
+* Two logging solutions (Shuffleboard recordings, on-RIO logging)
```

Comment:
Is "on-RIO logging" the DataLog+DataLogTool that is mentioned in the WPILib Blog as coming someday? With only a brief mention below it doesn't seem to be much integrated into this discussion of telemetry although I'd consider them the same (almost).
The blog also mentions:
• HAL telemetry
• Unified Offline Data Analysis Tool
Maybe?
• NetComm telemetry

Reply:
DataLog and DataLogTool were in the 2022.4.1 release and documented here: https://docs.wpilib.org/en/stable/docs/software/telemetry/datalog.html